src/mostlyai/qa/report.py

```diff
@@ -221,52 +221,41 @@ def report(
             statistics=statistics,
             workspace=workspace,
         )
-        on_progress(current=30, total=100)
+        on_progress(current=20, total=100)
 
-        # ensure that embeddings are all of equal size for a fair 3-way comparison
+        # ensure that embeddings are all equal size for a fair 3-way comparison
         max_sample_size_embeddings = min(
             max_sample_size_embeddings or float("inf"),
             syn_sample_size,
             trn_sample_size,
             hol_sample_size or float("inf"),
         )
-        # calculate embeddings
-        syn_embeds = calculate_embeddings(
-            pull_data_for_embeddings(
-                df_tgt=syn_tgt_data,
-                df_ctx=syn_ctx_data,
-                ctx_primary_key=ctx_primary_key,
-                tgt_context_key=tgt_context_key,
-                max_sample_size=max_sample_size_embeddings,
-            )
-        )
-        _LOG.info(f"calculated embeddings for synthetic {syn_embeds.shape}")
-        on_progress(current=50, total=100)
-        trn_embeds = calculate_embeddings(
-            pull_data_for_embeddings(
-                df_tgt=trn_tgt_data,
-                df_ctx=trn_ctx_data,
+
+        def _calc_pull_embeds(df_tgt: pd.DataFrame, df_ctx: pd.DataFrame, start: int, stop: int) -> np.ndarray:
+            strings = pull_data_for_embeddings(
+                df_tgt=df_tgt,
+                df_ctx=df_ctx,
                 ctx_primary_key=ctx_primary_key,
                 tgt_context_key=tgt_context_key,
                 max_sample_size=max_sample_size_embeddings,
             )
-        )
-        _LOG.info(f"calculated embeddings for training {trn_embeds.shape}")
-        on_progress(current=60, total=100)
+            # split into buckets for calculating embeddings to avoid memory issues and report continuous progress
+            buckets = np.array_split(strings, stop - start)
+            embeds = []
+            for i, bucket in enumerate(buckets, 1):
+                embeds += [calculate_embeddings(bucket.tolist())]
+                on_progress(current=start + i, total=100)
```

Comment:
not a big issue, but probably current=start + i + 1, 
since eg, 20:
we call: syn_embeds = _calc_pull_embeds(df_tgt=syn_tgt_data, df_ctx=syn_ctx_data, start=20, stop=40)
but 20 was already done above here: (on_progress(current=20), total=100)